src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryService.java

```diff
@@ -133,11 +211,50 @@ public Optional<TaskResource> getTask(String taskId,
             || taskId.isBlank()) {
             return Optional.empty();
         }
-        final Specification<TaskResource> taskResourceSpecification = TaskResourceSpecification
-            .buildSingleTaskQuery(taskId, accessControlResponse, permissionsRequired);
 
-        return taskResourceRepository.findOne(taskResourceSpecification);
+        SelectTaskResourceQueryBuilder selectQueryBuilder = new SelectTaskResourceQueryBuilder(entityManager);
+
+        final Predicate selectPredicate = TaskSearchQueryBuilder.buildSingleTaskQuery(
+            taskId,
+            accessControlResponse,
+            permissionsRequired,
+            selectQueryBuilder.builder,
+            selectQueryBuilder.root
+        );
+
+        selectQueryBuilder.where(selectPredicate).build().getResultList();
+
+        try {
+            return selectQueryBuilder
+                .where(selectPredicate)
+                .build()
+                .getResultList().stream()
+                .findFirst();
+        } catch (NoResultException ne) {
```

Comment:
New line following re-engineering for performance. Can we add unit test to cover this line (edge case negative scenarios)?